### DISCUSSION TITLE
AOSP 12.1 v416

### DISCUSSION THREAD

**Discussion by phhusson** [phhusson/treble_experimentations#2469] [General]
Still July security patch

- Fixes fingerprint enrolling crashing on gapps variant (only for 64bits, sorry, 32bits will still need to disable Android Setup app after setup wizard)
- Fixes the huge memory leak that made reboot necessary every few days

<hr /><em>This discussion was created from the release <a href='https://github.com/phhusson/treble_experimentations/releases/tag/v416'>AOSP 12.1 v416</a>.</em>

**Comment by ghost** [comment]


**Reply by enessmr** [reply]
is this supports redmi note 6 pro

**Comment by jwagnervaz** [comment]
Any fix for samsung offline charger ? the patch is here 
https://github.com/ponces/treble_build_pe/blob/twelve/patches/personal/common/platform_system_core/0003-fix-offline-charger-v7.patch

please fix this in the next update.

**Reply by phhusson** [reply]
There is no next update. I did this release only because I had critical bugs left.

**Comment by twindragon6** [comment]
The overlay for the OnePlus Nord N10 isn't working properly with the 11.0.7.BE86AA base firmware.
The main issue I'm having is the adaptive brightness does not work at all, everything else appears to work except DT2W.

I tried to build one following the instructions here:
https://github.com/phhusson/treble_experimentations/wiki/How-to-create-an-overlay%3F

This part could be cleaned up a bit and made more clear, some references, directory structure etc. I found it confusing to follow and I'm not even sure I did it correctly.
I was able to successfully build the APK however when I copied it to the device the package was corrupted somehow. It might be a signing issue? Maybe it wasn't built correctly? Dependencies broken?

> On your device, in adb or Termux app use this command: getprop ro.vendor.build.fingerprint
> 
> On PC, copy AndroidManifest.xml and replace in it:
> 
> package="me.phh.treble.overlay.brand.device"`
> 
> android:requiredSystemPropertyValue="+brand/device*"
> 
> Where brand and device for property value is what you got from that getprop command.
> 
> Then inside your device's directory, create a res folder and copy inside the xml folder (If there is no power_profile.xml in decompile system/framework/framework-res.apk and copy its res/xml/power_profile.xml into it) from apk decompiled by apktool. Then inside the res, also copy the values folder. Inside it, delete every file which isn't called arrays, bools, integers, strings (.xml).
> Build with your own PC
> 
> Install xmlstarlet if you don't have it by: sudo apt install xmlstarlet Then go into vendor_hardware_overlay/tests, open terminal and run: bash tests.sh
> 
> If you see lines like this defines a non-existing attribute APKTOOL_DUMMY_36 after test you need to delete this attribute from config.xml
> 
> As a result, you'll get a recommended priority level which you should replace in your AndroidManifest.xml. Inside the overlay.mk of main directory add your LOCAL_PACKAGE_NAME, in the alphabetic order.
> 
> Then if you want (but you should), go into vendor_hardware_overlay/build/ and run: bash build.sh It'll build you an overlay which you should put in /system/product/overlay/ on your device.
> 
> (if you see errors like this when you build overlays Error: No resource found that matches the given name (at 'storageDescription' with value '@string/storage_usb'). you need to remove this attributes from files)
> 


Using the source code here:
https://github.com/phhusson/vendor_hardware_overlay

And firmware here:
https://www.oneplus.com/support/softwareupgrade/details?code=PM1605857280263

Stock Files:
[OnePlus_Nord_N10.zip](https://github.com/phhusson/treble_experimentations/files/9996267/OnePlus_Nord_N10.zip)
[Stock_Overlay.zip](https://github.com/phhusson/treble_experimentations/files/9996287/Stock_Overlay.zip)
[dtbo.zip](https://github.com/phhusson/treble_experimentations/files/9997927/dtbo.zip)

**Reply by ercerd** [reply]
I think i have been there.
Did you look at the APK file permissions. set permission to rw-r--r-- / 0644.
Apk worked by setting file permisson.
I followed that guide and i also found it hard to follow.

**Reply by twindragon6** [reply]
I may have just built the thing wrong, I got it to build with a system recognizable apk once and for some reason can't reproduce the result. OnePlus file systems are bizarre to say the least. I'm wondering if it's just missing proprietary blobs that aren't included for some reason. Qualcomm kinda sucks when it comes to open and available support to end users. I don't care about the licensing, I just want it to work, borrowed code or not.

**Comment by innofeld** [comment]
Dear  Pierre-Hugues HUSSON 
Many thanks for this release, I need to know if we can install it on a device with Android 9.0 Pie ?
Regards
INNOFELD

**Reply by blessingmwiti** [reply]
As long as it has treble support

**Reply by D-e-vGoodify** [reply]
Yes you can with project treble being supported.
I'm currently using this GSI on my Android 9.0 device, my Tecno pouvoir 3 plus an arm32 VNDK 2.0 device successfully flashed the GSI.
The only issue I'm having is cellular data/mobile data connection, my device doesn't use VoLTE but rather 4g LTE and I've been trying to get my SIM connected but it doesn't work, even after installing IMS APK and setting up PHH IMS Apn, it shows the LTE in the signal but it isn't connected successfully, because of this i can't make and receive calls on the device.
Can anyone help 😕

**Comment by ghost** [comment]


**Reply by phhusson** [reply]
With the amount of information you gave, I can't.

**Comment by PembaTamang** [comment]
I am confused, I have aSM-J810G it runs an arm 64 bit processor, what files do I download how do I flash it.
https://www.getdroidtips.com/samsung-galaxy-j8-android-10/ this is what I am following and I can't find the **system.img** and **vbmeta.img** file mentioned in the 4th point under **Instructions to Install:**

**Comment by joranv1** [comment]
Is this memory leak fix already included in Android 13 GSI images?

**Comment by phhusson** [comment]
Yes

19 nov. 2022 14:43:00 Joran Vancoillie ***@***.***>:

**Comment by nation888** [comment]
Ref: Huawei 8X Max. Is it possible to copy system-quack-arm64-ab-floss.img  adaptive brightness overlay to Android 12?  I would like to copy the brightness overlay (can't seem to find) to this rom before flashing.  The Quack rom is the only rom that works for the adaptive brightness  with this phone.

**Comment by pranav-vaid** [comment]
Thank you @phhusson 
The memory leak issue was really a pain. 

Also, just to clarify, by this
> There is no next update. I did this release only because I had critical bugs left.

Do you mean that you have stopped working on Android 12.1 releases. Or you have completely stopped working on phh gsi's releases(which will mean that we will not see a Android 13 version of your GSI)?

Anyways, hats off and thanks to your amazing work here.

**Reply by phhusson** [reply]
I won't do the equivalent of current phh's gsi for Android 13. The "generic basis for all other ROMs" go into [TrebleDroid](https://github.com/TrebleDroid/). The AOSP  floss and gapps GSI ROMs currently have no alternative for Android 13. I might make a "phh custom rom" later, but that won't be aosp at all.

**Comment by ghost** [comment]
is this gonna be the last phh release?

**Reply by pranav-vaid** [reply]
Looks like it.

**Comment by BieGabs** [comment]
does anyone know how to get past this "32bit still need to disable android setup app after setup wizard)" problem?? Can anyone tell me how to resolve this?

**Reply by D-e-vGoodify** [reply]
If you're trying to skip Android setup wizard for arm32 device. The best way is to tap the edges of the screen in a clockwise motion, so tap:
1. Top left edge
2. Top right edge
3. Bottom right edge
4. Bottom left edge.
 That should now show a message of skipped setup wizard 👍

**Comment by GempitaRizki** [comment]
sdk version ?

**Comment by nation888** [comment]
Works perfectly on the Oukitel K9 thanks Phusson

**Comment by mrvanes** [comment]
If I wanted to try for a vanilla as possible Android 13 release that also has the patches to support my Jelly2 like this one, including gapps or at least the android play store, where to go?

**Reply by JonnyTech** [reply]
https://github.com/TrebleDroid/treble_experimentations/releases

**Reply by mrvanes** [reply]
Would it possible to flash over this AOSP 12.1 v414 without wiping?

**Reply by JonnyTech** [reply]
Try it `¯\_(ツ)_/¯`

**Comment by devrieshh** [comment]


**Reply by GempitaRizki** [reply]
have you tried to check on the icmmc section, I think there is a problem on the hardware side
cmiiw

**Reply by devrieshh** [reply]
you mean vendor.img

**Reply by GempitaRizki** [reply]
jst christ

**Comment by Edilson22659** [comment]
Oi eu instalei essa GSi No Meu LG K12 Max ela é perfeita pegou de boa mas eu achei um bug que é a impressão digital não pegar, meu LG K12 Max a arquitetura dele é arm32

**Comment by dev-eof** [comment]
Help, i installed this v416 release and the headphones has an issue. Both the headphones and the speaker work and even doesn't show it is detecting the headset. I've tried those settings in Phh Treble app and still doesn't fix the issue. Help?

**Comment by Mburu254ke** [comment]
my mobile data won't work.
wifi works fine
stk won't work too

**Reply by blessingmwiti** [reply]
In phh treble settings, under ims features, click everything, that is, create ims apn, install ims apk, request ims network and force the presence of 4G calling. Hope this solves you mobile data issue

**Comment by dev-eof** [comment]
About the wired headphones issue...on lineage 20 gsi. There was this
time it detected I don't how. Anyone with a clue?

**Comment by ace-kun-dev** [comment]
Hi, im wanting to access usb otg in this rom (my device is Tecno Spark 7t and i successfully flashed the img_of this rom into the system partition) i just want to know if its possible in this rom, thanks : )

**Reply by phhusson** [reply]
Try this: https://github.com/phhusson/treble_experimentations/issues/1006#issuecomment-565382616

**Reply by ace-kun-dev** [reply]
Thanks, i clicked the link then scrolled down, and fount a solution. the otg-enabler apk file. Thank you very much

**Comment by fanuelberhanukebe** [comment]
Android is not detecting my earphone(cable) jack Device ZTE Blade 20 Smart

**Comment by fanuelberhanukebe** [comment]
please help not detecting earphone not wireless, and how can I fix null IMEI

**Reply by dev-eof** [reply]
If it's Unisoc, speculation has it there isn't a solution for headphones issue. For IMEI you can try an IMEI writer

**Reply by phhusson** [reply]
If it works on stock rom's system.img, there is always a solution.

**Reply by dev-eof** [reply]
Can copying the audio policy xml files from stock help?

**Reply by phhusson** [reply]
That's unlikely but you can try

**Reply by fanuelberhanukebe** [reply]
It works on the stock rom It was Android 9, I've also tried your Android 10,11 but It is the same earphone jack don't work but The rest is fine THANK YOU👍

**Reply by dev-eof** [reply]
Try this for 13
https://sourceforge.net/projects/andyyan-gsi/files/lineage-20-td/

**Reply by fanuelberhanukebe** [reply]
Okay I will try but how can I backup my apps

**Reply by fanuelberhanukebe** [reply]
I've tried pixel experience unofficial havoc os but the os don't start my phone boots shows the ZTE logo then the havoc or googe(pixel) logo then It's stuck there hope this works Thank you

**Reply by fanuelberhanukebe** [reply]
I have another question I flash the rom with fastboot I've tried twrp It says device locked input password entered my lock screen It didn't work but it allows to modify, In install menu It asks me to change my storage format which is emmc I searched the internet and changed it to ext/4 but my phone boots to recovery mode only even after I flashed Rom, before changing the storage format my storage says 0mb

**Comment by dev-eof** [comment]
For backup check out this thread https://forum.xda-developers.com/t/best-apps-for-backups-speed-changing-rom.4062857/

For flashing try using bootloader, I don't know about twrp

**Reply by fanuelberhanukebe** [reply]
my earphone is now detected I enabled settings phh treble settings Misc features and I ticked what says detect audio kinda thing THANK YOU FOR HELPING

**Reply by dev-eof** [reply]
That's great, i decided to use BT pods instead. Unisoc devices don't get along properly with GSI, many users reported that (https://www.hovatek.com/forum/thread-39451.html)

**Reply by fanuelberhanukebe** [reply]
Have you tried what what I did?

**Reply by dev-eof** [reply]
Yeah, no alternatives i guess

**Reply by fanuelberhanukebe** [reply]
how can I install magisk in this rom it doesn't work by patching the boot img

**Comment by Nemethze** [comment]
Hi guys, when will android 13 arrive?

**Reply by JonnyTech** [reply]
Android 13 has been out for a year - Android 14 will soon be here...

**Comment by Yuri29710** [comment]
Phhusson?  why not do one last serious update?  I like most of your gsis but there is a super critical problem that when you lock the screen the touch stops working this makes it almost impossible to use the gsi forcing you to restart or leave the screen on infinitely I know you solved this in the versions of the gsi Android 13 and wonder why not add google apps in your android 13 gsis?  it's complicated to put apps like apkpure and etc on the memory card I realized that without the google play store and google play services it's almost impossible to look for an app of interest and you should change that ridiculous browser called WebView Shell that doesn't allow installation of any app on the web sorry for this text :/

**Reply by spectrum70** [reply]
"Phhusson? why not do one last serious update? "  this is a great way to produce "no more releases".

**Comment by fanuelberhanukebe** [comment]
how can I install magisk in this rom it doesn't work by patching the boot img?

**Comment by dev-eof** [comment]
Install bgN version, maybe your current one has an inbuilt superuser
which conflicts with magisk

**Reply by GempitaRizki** [reply]
jsus crist , flash and re ubl and re root your devices

**Comment by mahmud3301** [comment]
I can't understand which one for tecno pop 2 please help me

**Comment by devrieshh** [comment]
I have a Cubot king kong mini 3 and the system can`t be written on it i say written because the whole procedure works, system flashed with no errors,when i do a fastboot -w  afterwords  there is almost no outcome,only that it takes 0sec  for the job.
than it wont boot so i have to flash the stock rom with SP_Flash_Tool

on my older Cubot king kong mini2 pro the flash procedure worked 

Is it possible that it has anything to do with unlocking the bootloader? It says it doesn't,but after rebooting it shows all the signs of unlock

**Comment by EX-maker** [comment]
Hello👋. Is it possible to install this firmware on Tecno spark 8c?

**Reply by devrieshh** [reply]
see what the Treble Info.app says

**Reply by EX-maker** [reply]
Writes that Project Treble is supported on your device

**Reply by devrieshh** [reply]
Then it says the one you need , probably a,and not newer arm64

**Reply by EX-maker** [reply]
There are arm64 files there, should I download them all?

**Reply by devrieshh** [reply]
i guess you need a 32bit system(older,android 8.1 or so see treble app)so you could try https://github.com/phhusson/treble_experimentations/releases/download/v416/system-squeak-arm32_binder64-ab-gogapps.img.xz

**Reply by EX-maker** [reply]
I looked and I have a 64-bit system, which file should I download then?

**Reply by EX-maker** [reply]
And I have 11 android costs

**Reply by devrieshh** [reply]
I guess the arm64 ones

**Reply by devrieshh** [reply]
https://github.com/phhusson/treble_experimentations/releases/download/v416/system-squeak-arm64-ab-gapps.img.xz

**Reply by EX-maker** [reply]
Thank you so much for the answers 😁

**Reply by EX-maker** [reply]
This custom firmware is on a naked android, so it weighs so little?

**Comment by Cleo-cmyk** [comment]
I need help, my phone is a galaxy a31, I installed this gsi but when I install magisk and click on "direct install" it gives signature failure, I use version 24.3 of magisk.

**Comment by ruudboy** [comment]
Aucune option d'appel wifi dans "Appels et SMS" ou dans Paramètre compte téléphonique/paramètre sur "system-squeak-arm64-ab-gapps" [AOSP 12.1 v416] (https://github.com/phhusson/treble_experimentations/releases/tag/v416) flash phone Cubot Quest 
Comment obtenir l'option appel wifi ? l'option est elle disponible sur cette version 12.1 v416 ?
Merci

No wifi calling option in "Calls & SMS" or in Phone account setting/setting on "system-squeak-arm64-ab-gapps" [AOSP 12.1 v416] (https://github.com/phhusson/treble_experimentations/releases /tag/v416) flash phone Cubot Quest
How to get the wifi calling option? Is the option available on this version 12.1 v416?
THANKS

**Comment by sieger707** [comment]
I am new here based on [this XDA thread](https://xdaforums.com/t/root-you-android.4203297/) AOSP by 
phhusson are rooted. Please confirm. If not I am stuck because I have an  AB phone non standard &  CCP  type    with no TWRP available and magisk patching method gives a boot loop.  Can someone please clarify / help .

**Comment by devrieshh** [comment]
maybe a factory reset helps

**Comment by rogers1106** [comment]
Hello, I know I'm late but can I still use the last version (12.1 v416) on a Oukitel WP6?

Thanks a lot.

**Comment by devrieshh** [comment]
Nothing has changed on the last version,so

**Reply by rogers1106** [reply]
so ...?

**Comment by devrieshh** [comment]
So it should, when the device is treble compatibel 
Treble info app

**Comment by rikemade** [comment]
It gives an authorization required error while downloading the file.
